SSLHandshakeException when submitting images

Several users found an SSL handshake issue.

```
(Fiji Is Just) ImageJ 2.1.0/1.53c; Java 1.8.0_66 [64-bit]; Windows 10 10.0; 119MB of 18258MB (<1%)
 
javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

	at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)

	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)

	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302)

	at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296)

	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1509)

	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216)

	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979)

	at sun.security.ssl.Handshaker.process_record(Handshaker.java:914)

	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)

	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)

	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)

	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)

	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:367)

	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:325)

	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:197)

	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:249)

	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:108)

	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:76)

	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:245)

	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)

	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)

	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:96)

	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)

	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)

	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)

	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)

	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)

	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:197)

	at okhttp3.internal.connection.RealCall.execute(RealCall.kt:148)

	at org.vanvalenlab.KioskHttpClient.sendHttpRequest(KioskHttpClient.java:35)

	at org.vanvalenlab.KioskHttpClient.getJobTypes(KioskHttpClient.java:126)

	at org.vanvalenlab.KioskJobManager.selectJobType(KioskJobManager.java:38)

	at org.vanvalenlab.ImageJobManager.run(ImageJobManager.java:29)

	at ij.IJ.runUserPlugIn(IJ.java:235)

	at ij.IJ.runPlugIn(IJ.java:198)

	at ij.Executer.runCommand(Executer.java:150)

	at ij.Executer.run(Executer.java:68)

	at java.lang.Thread.run(Thread.java:745)

Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

	at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387)

	at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:292)

	at sun.security.validator.Validator.validate(Validator.java:260)

	at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)

	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)

	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)

	at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1491)

	... 33 more

Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

	at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:146)

	at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:131)

	at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)

	at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382)

	... 39 more
```

--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@ The `kiosk-imageJ-plugin` is a ImageJ 1.x plugin for easily processing images wi
 
 ## How to install
 
-1. Download the [latest JAR file](https://github.com/vanvalenlab/kiosk-imageJ-plugin/releases/download/0.3.1/Kiosk_ImageJ-0.3.1.jar).
+1. Download the [latest JAR file](https://github.com/vanvalenlab/kiosk-imageJ-plugin/releases/download/0.3.2/Kiosk_ImageJ-0.3.2.jar).
 2. Open ImageJ.
 3. Navigate to Plugins > Install...
 4. Select the downloaded JAR file.

--- a/pom.xml
+++ b/pom.xml
@@ -13,7 +13,7 @@
 
   <groupId>com.vanvalenlab</groupId>
   <artifactId>Kiosk_ImageJ</artifactId>
-  <version>0.3.1</version>
+  <version>0.3.2</version>
   <packaging>jar</packaging>
 
   <name>DeepCell Kiosk ImageJ Plugin</name>

--- a/src/main/java/org/vanvalenlab/KioskHttpClient.java
+++ b/src/main/java/org/vanvalenlab/KioskHttpClient.java
@@ -6,8 +6,10 @@ import org.vanvalenlab.responses.*;
 import com.google.gson.Gson;
 import okhttp3.*;
 
+import javax.net.ssl.*;
 import java.io.File;
 import java.io.IOException;
+import java.util.concurrent.TimeUnit;
 
 public class KioskHttpClient {
 
@@ -20,9 +22,53 @@ public class KioskHttpClient {
      * @param host The DeepCell Kiosk host.
      */
     public KioskHttpClient(String host) {
-        this.client = new OkHttpClient();
+        this.client = this.generateDefaultOkHttp();
         this.g = new Gson();
         this.host = host;
+    }
+
+    private static OkHttpClient generateDefaultOkHttp() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        try {
+            // Create a trust manager that does not validate certificate chains
+            final TrustManager[] trustAllCerts = new TrustManager[]{
+                new X509TrustManager() {
+                    @Override
+                    public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+                    }
+
+                    @Override
+                    public void checkServerTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+                    }
+
+                    @Override
+                    public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                        return new java.security.cert.X509Certificate[]{};
+                    }
+                }
+            };
+
+            // Install the all-trusting trust manager
+            final SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+            // Create an ssl socket factory with our all-trusting manager
+            final SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();
+
+            builder.sslSocketFactory(sslSocketFactory, (X509TrustManager) trustAllCerts[0]);
+            builder.hostnameVerifier(new HostnameVerifier() {
+                @Override
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        builder.connectTimeout(60, TimeUnit.SECONDS)
+            .readTimeout(60, TimeUnit.SECONDS)
+            .writeTimeout(60, TimeUnit.SECONDS)
+            .retryOnConnectionFailure(true);
+        return builder.build();
     }
 
     /**

